homeassistant/components/sensor/mvglive.py

```diff
@@ -0,0 +1,152 @@
+# A typo in the file name of the PyPI version prevents installation from PyPI
+REQUIREMENTS = ["PyMVGLive==1.1.3"]
+ICON = 'mdi:bus'
+
+# Return cached results if last scan was less then this time ago.
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=15)
+
+CONF_STATION = 'station'
+CONF_DEST = 'destination'
+CONF_LINE = 'line'
+CONF_OFFSET = 'offset'
+CONF_UBAHN = 'ubahn'
+CONF_TRAM = 'tram'
+CONF_BUS = 'bus'
+CONF_SBAHN = 'sbahn'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_STATION): cv.string,
+    vol.Optional(CONF_DEST, default=None): cv.string,
+    vol.Optional(CONF_LINE, default=None): cv.string,
+    vol.Optional(CONF_OFFSET, default=0): cv.positive_int,
+    vol.Optional(CONF_UBAHN, default=True): cv.boolean,
+    vol.Optional(CONF_TRAM, default=True): cv.boolean,
+    vol.Optional(CONF_BUS, default=True): cv.boolean,
+    vol.Optional(CONF_SBAHN, default=True): cv.boolean,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the MVG Live Sensor."""
+    station = config.get(CONF_STATION)
+    destination = config.get(CONF_DEST)
+    line = config.get(CONF_LINE)
+    offset = config.get(CONF_OFFSET)
+    ubahn = config.get(CONF_UBAHN)
+    tram = config.get(CONF_TRAM)
+    bus = config.get(CONF_BUS)
+    sbahn = config.get(CONF_SBAHN)
+
+    add_devices([MVGLiveSensor(station, destination, line,
+                               offset, ubahn, tram, bus, sbahn)], True)
+
+
+# pylint: disable=too-few-public-methods
+class MVGLiveSensor(Entity):
+    """Implementation of an MVG Live sensor."""
+
+    def __init__(self, station, destination, line,
+                 offset, ubahn, tram, bus, sbahn):
+        """Initialize the sensor."""
+        self._station = station
+        self._destination = destination
+        self._line = line
+        self.data = MVGLiveData(station, destination, line,
+                                offset, ubahn, tram, bus, sbahn)
+        self._state = STATE_UNKNOWN
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        # e.g.
+        # 'Hauptbahnhof (S1)'
+        # 'Hauptbahnhof-Marienplatz'
+        # 'Hauptbahnhof-Marienplatz (S1)'
+        namestr = self._station
+        if self._destination:
+            namestr = namestr + '-' + self._destination
+        if self._line:
+            namestr = namestr + ' (' + self._line + ')'
+        return namestr
+
+    @property
+    def icon(self):
+        """Return the icon for the frontend."""
+        return ICON
+
+    @property
+    def state(self):
+        """Return the departure time of the next train."""
+        return self._state
+
+    @property
+    def state_attributes(self):
+        """Return the state attributes."""
+        if self.data.departures:
+            return self.data.departures[0]
+
+    def update(self):
+        """Get the latest data and update the state."""
+        self.data.update()
+        if not self.data.departures:
```

Comment:
This case can never happen because you always have an empty dictionary in your list, meaning it will always test truthy. (only an empty list is False)